src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTests.cs

```diff
@@ -16261,13 +16294,302 @@ .maxstack  2
+{
+  // Code size       70 (0x46)
+  .maxstack  4
+  .locals init (int V_0,
+                System.Span<int> V_1,
+                System.Span<int> V_2)
+  IL_0000:  ldc.i4.2
+  IL_0001:  stloc.0
+  IL_0002:  ldc.i4.8
+  IL_0003:  conv.u
+  IL_0004:  localloc
+  IL_0006:  dup
+  IL_0007:  ldc.i4.4
+  IL_0008:  stind.i4
+  IL_0009:  dup
+  IL_000a:  ldc.i4.4
+  IL_000b:  add
+  IL_000c:  ldc.i4.5
+  IL_000d:  stind.i4
+  IL_000e:  ldc.i4.2
+  IL_000f:  newobj     ""System.Span<int>..ctor(void*, int)""
+  IL_0014:  stloc.1
+  IL_0015:  ldc.i4.s   12
+  IL_0017:  conv.u
+  IL_0018:  localloc
+  IL_001a:  dup
+  IL_001b:  ldc.i4.1
+  IL_001c:  stind.i4
+  IL_001d:  dup
+  IL_001e:  ldc.i4.4
+  IL_001f:  add
+  IL_0020:  ldloca.s   V_1
+  IL_0022:  call       ""int System.Span<int>.Length.get""
+  IL_0027:  stind.i4
+  IL_0028:  dup
+  IL_0029:  ldc.i4.2
+  IL_002a:  conv.i
+  IL_002b:  ldc.i4.4
+  IL_002c:  mul
+  IL_002d:  add
+  IL_002e:  ldc.i4.3
+  IL_002f:  stind.i4
+  IL_0030:  ldc.i4.3
+  IL_0031:  newobj     ""System.Span<int>..ctor(void*, int)""
+  IL_0036:  stloc.2
+  IL_0037:  ldloc.0
+  IL_0038:  ldloca.s   V_2
+  IL_003a:  call       ""int System.Span<int>.Length.get""
+  IL_003f:  mul
+  IL_0040:  call       ""void System.Console.WriteLine(int)""
+  IL_0045:  ret
+}");
+            comp = CreateCompilationWithMscorlibAndSpan(source, TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: "6", verify: Verification.Fails).VerifyIL("Test.Main", @"
+{
+  // Code size       74 (0x4a)
+  .maxstack  4
+  .locals init (int V_0, //z
+                int V_1,
+                System.Span<int> V_2,
+                System.Span<int> V_3)
+  IL_0000:  nop
+  IL_0001:  ldc.i4.2
+  IL_0002:  stloc.0
+  IL_0003:  ldloc.0
+  IL_0004:  stloc.1
+  IL_0005:  ldc.i4.8
+  IL_0006:  conv.u
+  IL_0007:  localloc
+  IL_0009:  dup
+  IL_000a:  ldc.i4.4
+  IL_000b:  stind.i4
+  IL_000c:  dup
+  IL_000d:  ldc.i4.4
+  IL_000e:  add
+  IL_000f:  ldc.i4.5
+  IL_0010:  stind.i4
+  IL_0011:  ldc.i4.2
+  IL_0012:  newobj     ""System.Span<int>..ctor(void*, int)""
+  IL_0017:  stloc.2
+  IL_0018:  ldc.i4.s   12
+  IL_001a:  conv.u
+  IL_001b:  localloc
+  IL_001d:  dup
+  IL_001e:  ldc.i4.1
+  IL_001f:  stind.i4
+  IL_0020:  dup
+  IL_0021:  ldc.i4.4
+  IL_0022:  add
+  IL_0023:  ldloca.s   V_2
+  IL_0025:  call       ""int System.Span<int>.Length.get""
+  IL_002a:  stind.i4
+  IL_002b:  dup
+  IL_002c:  ldc.i4.2
+  IL_002d:  conv.i
+  IL_002e:  ldc.i4.4
+  IL_002f:  mul
+  IL_0030:  add
+  IL_0031:  ldc.i4.3
+  IL_0032:  stind.i4
+  IL_0033:  ldc.i4.3
+  IL_0034:  newobj     ""System.Span<int>..ctor(void*, int)""
+  IL_0039:  stloc.3
+  IL_003a:  ldloc.1
+  IL_003b:  ldloca.s   V_3
+  IL_003d:  call       ""int System.Span<int>.Length.get""
+  IL_0042:  mul
+  IL_0043:  call       ""void System.Console.WriteLine(int)""
+  IL_0048:  nop
+  IL_0049:  ret
 }");
         }
```

Comment:
>} [](start = 8, length = 1)

Might it be worth testing passing a stackalloc to a params parameter (`params Span<T>[] p`)?

Reply:
It would be an error to use `Span<T>` as the element type of an array.  Similarly, `stackalloc` is not a constant.